hashicorp/boundary-ui
# Icu 12933 updates to target model to support aliasing

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JIRA_TICKET_NUMBER)

## Description

This PR adds support to the target model to allow the creation of aliases within targets. 

NOTE: After you create it, you cannot see the aliases in the UI. This will be done in a separate PR. 



## Screenshots (if appropriate):


https://github.com/hashicorp/boundary-ui/assets/15043878/6cf737d2-d431-466a-a036-670d6d4e77b5



![target](https://github.com/hashicorp/boundary-ui/assets/15043878/4341ed10-84fc-4b8f-9780-caab10e7b4ea)


## How to Test



1. Get the latest boundary main, `make install` and  start the controller 
2. go to targets `/targets` within a project
3. you should only be able to see the option to create aliases with the target creation form if the user has access to create aliases 
4. Input name, select TCP target, input default port, and enter 0 or more aliases. 
5. Hit submit and save successfully.

**Note**: the above steps are for TCP targets; if you want to test SSH targets, you should use enterprise binary.  



:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:
